bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/handler/SatelStateThingHandler.java

```diff
@@ -86,16 +86,22 @@ public void incomingEvent(SatelEvent event) {
             }
             for (Channel channel : getThing().getChannels()) {
                 ChannelUID channelUID = channel.getUID();
-                StateType stateType = getStateType(channelUID.getId());
-                if (stateType != null && stateEvent.hasDataForState(stateType)) {
-                    int bitNbr = thingConfig.getId() - 1;
-                    boolean invertState = thingConfig.isStateInverted();
-                    updateSwitch(channelUID, stateEvent.isSet(stateType, bitNbr) ^ invertState);
+                if (isLinked(channel.getUID())) {
+                    StateType stateType = getStateType(channelUID.getId());
+                    if (stateType != null && stateEvent.hasDataForState(stateType)) {
+                        int bitNbr = getStateBitNbr(stateType);
+                        boolean invertState = thingConfig.isStateInverted();
+                        updateSwitch(channelUID, stateEvent.isSet(stateType, bitNbr) ^ invertState);
+                    }
                 }
             }
         }
     }
 
+    protected int getStateBitNbr(StateType stateType) {
```

Comment:
Please add JavaDoc to all non-private methods

Reply:
Done.